### PR TITLE
Remove moment dependency

### LINKED ISSUE #28: Switch to dateFns or other
Use of Moment is discouraged by the project itself https://momentjs.com/docs/#/-project-status/
switching to something like https://date-fns.org/ would reduce package sizes for those using this project.

### PR DESCRIPTION
**Pull request by cihancimen**
Removes moment. There were a few critical date related operations provided by moment. They are now moved to data_utils.ts. It uses javascript's native Date type. Functions in date_utils are expected to work in local timezone, but should handle Daylight Savings Time within the timezone.

resolves #28

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,9 +41,7 @@
     "tslint-config-prettier": "^1.18.0",
     "typescript": "^3.8.3"
   },
-  "dependencies": {
-    "moment": "^2.25.3"
-  },
+  "dependencies": {},
```
Comment (sametaylak) [inline package.json:44]: https://www.youtube.com/watch?v=3WAOxKOmR90

File: src/date_utils.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+export function getDayOfYear(date: Date): number {
+    const millisecondsInDay = 1000 * 60 * 60 * 24
+    return (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) - Date.UTC(date.getFullYear(), 0, 0)) / millisecondsInDay;
+}
+
+export function toFormattedString(date: Date): string {
+    return new Date(date.getTime() - (date.getTimezoneOffset() * 60000))
+        .toISOString()
+        .split("T")[0];
```
Comment (sametaylak) [inline src/date_utils.ts:67]: I couldn't understand what that means
```javascript
(date.getTimezoneOffset() * 60000)
```

**Review by sametaylak — APPROVED** [review]
Have a minor comment on code readability

File: src/date_utils.ts (hunk 2 above)
Comment (cihancimen) [reply src/date_utils.ts:67]: This function returns the date string in `YYYY-MM-DD` format. To formatting we use `toISOString()` which returns a string in `YYYY-MM-DDTHH:mm:ss.sssZ` format. Then we split by T and take first part.
However, there is a problem. `toISOString()` result is not in Date object's timezone. [Returned string's timezone is always zero UTC offset](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString). That is why we need to construct a new Date in a timezone with zero UTC offset. We get current timezoneoffset by `date.getTimezoneOffset()` which is in minutes and multiply it with 60 * 1000 to convert to milliseconds.

**Comment by burakgoynuk** [comment]
@cihancimen @sametaylak should we release this? I made sense to me as well :)

**Pull request merged**